### ISSUE TITLE
Bad Request - Error in query syntax while publishing using ALOps ExtensionAPI

### ISSUE THREAD

**Issue by VolkovYuVa** [HodorNV/ALOps#782]
**Description**
An error occurs while trying to publish an .app artifact to Dynamics BC 365.
It seems that for publishing the extension is trying to reach the \*\*\*/v2.0/\*\*\*/sandbox/api/microsoft/automation/v2.0/companies(5acb1e92-575c-ee11-8def-6045bd0fe76f)/extensionUpload(0)/**content** but it should have reached the extensionContent instead according to this page:
https://learn.microsoft.com/ru-ru/dynamics365/business-central/dev-itpro/administration/api/dynamics_extensionupload_update
<img width="621" alt="image" src="https://github.com/user-attachments/assets/07a573f2-c52e-454b-8698-6df5f849bce7">

**The used yaml**
```
steps:
- task: Hodor.hodor-alops.ALOpsExtensionAPI.ALOpsExtensionAPI@1
  displayName: 'ALOps Extension API'
  inputs:
    interaction: publish
    api_endpoint: '$(Global-BC-Resource)/v2.0/$(Global-Cloud-TenantID)/$(Release.EnvironmentName)/api'
    apiversion: v2.0
    azure_tenant_id: '$(Global-Cloud-TenantID)'
    azure_app_client_id: '$(Global-BC-ClientID)'
    azure_app_client_secret: '$(Global-BC-Secret)'
    checksecondsdelay: 600
    replacepackageid: true
```

**The output**
```
2024-08-08T12:09:14.3482606Z ==============================================================================
2024-08-08T12:09:14.3482779Z Task         : ALOps Extension API
2024-08-08T12:09:14.3482858Z Description  : Get/Publish extensions with the Business Central API
2024-08-08T12:09:14.3482985Z Version      : 1.465.6167
2024-08-08T12:09:14.3483117Z Author       : Hodor
2024-08-08T12:09:14.3484222Z Help         : Get/Publish extensions with the Business Central API.
2024-08-08T12:09:14.3484487Z ==============================================================================
2024-08-08T12:09:15.8460525Z *** Validate configuration
2024-08-08T12:09:16.0191300Z *** Task Inputs:
2024-08-08T12:09:16.0229727Z 
2024-08-08T12:09:16.0357975Z name                                                                                                              value
2024-08-08T12:09:16.0359453Z ----                                                                                                              -----
2024-08-08T12:09:16.0361217Z usedocker                                                                                                         False
2024-08-08T12:09:16.0362783Z fixed_tag                                                                                                              
2024-08-08T12:09:16.0364530Z interaction                                                                                                       batch
2024-08-08T12:09:16.0392615Z api_endpoint                          ...esscentral.dynamics.com//v2.0/***/sandbox/api
2024-08-08T12:09:16.0393389Z dev_endpoint                                                                                                           
2024-08-08T12:09:16.0396065Z dev_schemeupdatemode                                                                                        synchronize
2024-08-08T12:09:16.0397315Z apiversion                                                                                                         v2.0
2024-08-08T12:09:16.0399062Z authentication                                                                                                    oauth
2024-08-08T12:09:16.0400362Z azure_tenant_id                                                                    ***
2024-08-08T12:09:16.0403514Z azure_app_client_id                                                                ***
2024-08-08T12:09:16.0404152Z azure_app_client_secret                                                        ***
2024-08-08T12:09:16.0405679Z azure_app_client_certificate                                                                                           
2024-08-08T12:09:16.0407465Z azure_app_client_certificate_password                                                                                  
2024-08-08T12:09:16.0408878Z username                                                                                                               
2024-08-08T12:09:16.0411053Z password                                                                                                               
2024-08-08T12:09:16.0412568Z bccompany                                                                          5acb1e92-575c-ee11-8def-6045bd0fe76f
2024-08-08T12:09:16.0415670Z artifact_path                                                                                                 D:\a\r1\a
2024-08-08T12:09:16.0420172Z artifact_filter                                                                                                   *.app
2024-08-08T12:09:16.0422717Z showdeploymentstatus                                                                                               True
2024-08-08T12:09:16.0423946Z checksecondsdelay                                                                                                   600
2024-08-08T12:09:16.0425604Z maxtries                                                                                                             20
2024-08-08T12:09:16.0427640Z replacepackageid                                                                                                   True
2024-08-08T12:09:16.0428901Z blocksymbolsonly                                                                                                  False
2024-08-08T12:09:16.0430693Z pwsh                                                                                                              False
2024-08-08T12:09:16.0430928Z 
2024-08-08T12:09:16.0460114Z 
2024-08-08T12:09:16.0475244Z 
2024-08-08T12:09:16.0736647Z False
2024-08-08T12:09:16.0986644Z *** For documentation, please visit   : https://www.alops.be/documentation
2024-08-08T12:09:16.0992055Z 
2024-08-08T12:09:16.4956953Z *** ALOps License:
2024-08-08T12:09:16.4982973Z   * Trial Key  : 635d6029-a090-4cf8-9c99-0711960cfeee
2024-08-08T12:09:16.5008393Z   * Licensed To: ALOps Trial [29/31] days remaining. Trial ends: Saturday, 07 September 2024
2024-08-08T12:09:16.5015883Z 
2024-08-08T12:09:16.6577384Z *** Imported required PS-Functions
2024-08-08T12:09:16.6600974Z *** Parse Task Parameters
2024-08-08T12:09:16.9250205Z *** Creating API Connector with endpoint [***/v2.0/***/sandbox/api]
2024-08-08T12:09:16.9822301Z *** API Version: v2.0
2024-08-08T12:09:16.9849309Z *** Interaction Mode: batch
2024-08-08T12:09:16.9887694Z *** API Authentication: Using Azure OAuth Authentication [***]
2024-08-08T12:09:16.9918589Z *** Get OAuth token for tenant [***]
2024-08-08T12:09:17.0317672Z *** Communication: Service-2-Service (Client Secret)
2024-08-08T12:09:17.0354086Z ##[command]Invoke-RestMethod -Method Post -Uri "https://login.microsoftonline.com/***/oauth2/token"
2024-08-08T12:09:17.2569983Z *** BC Authentication: Token [Bearer]
2024-08-08T12:09:17.2640607Z *** List Companies
2024-08-08T12:09:17.2751148Z ##[command]Invoke-RestMethod -Method Get -Uri '***/v2.0/***/sandbox/api/microsoft/automation/v2.0/companies'
2024-08-08T12:09:17.6082595Z 
2024-08-08T12:09:17.6115409Z 
2024-08-08T12:09:17.6116682Z id                : 5acb1e92-575c-ee11-8def-6045bd0fe76f
2024-08-08T12:09:17.6117443Z systemVersion     : 24.3.21374.21517
2024-08-08T12:09:17.6118293Z timestamp         : 100290
2024-08-08T12:09:17.6129090Z name              : ****
2024-08-08T12:09:17.6129761Z displayName       : ****
2024-08-08T12:09:17.6130275Z businessProfileId : 
2024-08-08T12:09:17.6130801Z systemCreatedAt   : 2023-09-26T10:29:28.86Z
2024-08-08T12:09:17.6131309Z systemCreatedBy   : 84118961-da54-4f85-b360-78ae8dba42c6
2024-08-08T12:09:17.6131838Z systemModifiedAt  : 2023-09-26T10:29:28.86Z
2024-08-08T12:09:17.6132435Z systemModifiedBy  : 84118961-da54-4f85-b360-78ae8dba42c6
2024-08-08T12:09:17.6133108Z 
2024-08-08T12:09:17.6133369Z 
2024-08-08T12:09:17.6133641Z 
2024-08-08T12:09:17.6226003Z *** Determine Company with filter [5acb1e92-575c-ee11-8def-6045bd0fe76f]
2024-08-08T12:09:17.6363947Z *** Selected Company: **** [5acb1e92-575c-ee11-8def-6045bd0fe76f]
2024-08-08T12:09:17.6379795Z ##[command]Invoke-RestMethod -Method Get -Uri '***/v2.0/***/sandbox/api/microsoft/automation/v2.0/companies(5acb1e92-575c-ee11-8def-6045bd0fe76f)/extensions'
2024-08-08T12:09:17.9711045Z *** Installed Extensions:
......
...... (There are a list of installed extensions)
......
......
......
2024-08-08T12:09:18.0920220Z 
2024-08-08T12:09:18.0920508Z 
2024-08-08T12:09:18.0922842Z 
2024-08-08T12:09:18.0931021Z 
2024-08-08T12:09:18.0947423Z *** Published Extensions (not installed):
2024-08-08T12:09:18.1012681Z 
2024-08-08T12:09:18.1186147Z *** Import NAV/BC Management DLL's [False]
2024-08-08T12:09:18.1607247Z *** ServiceTier not specified, loading default assemblies.
2024-08-08T12:09:18.1716989Z *** Default Assemblies not found. Load via BC-Artifacts
2024-08-08T12:09:18.1854983Z *** Install/Update BCContainerHelper [https://github.com/microsoft/navcontainerhelper]
2024-08-08T12:09:18.1908498Z *** Check BcContainerHelper
2024-08-08T12:10:54.8448206Z *** Install BcContainerHelper
2024-08-08T12:11:11.9366003Z *** Load BcContainerHelper
2024-08-08T12:11:12.3517271Z BcContainerHelper version 6.0.19
2024-08-08T12:11:15.3261172Z BC.HelperFunctions emits usage statistics telemetry to Microsoft
2024-08-08T12:11:15.4836741Z Running on Windows, PowerShell 5.1.17763.6054
2024-08-08T12:11:18.8599367Z 
2024-08-08T12:11:18.8600176Z ModuleType Version    Name                                ExportedCommands                                             
2024-08-08T12:11:18.8600442Z ---------- -------    ----                                ----------------                                             
2024-08-08T12:11:18.8600792Z Script     6.0.19     BcContainerHelper                   {Add-FontsToBcContainer, Add-GitToAlProjectFolder, AddTele...
2024-08-08T12:11:18.8600922Z 
2024-08-08T12:11:18.8600994Z 
2024-08-08T12:11:18.8601038Z 
2024-08-08T12:11:18.8601083Z 
2024-08-08T12:11:18.8601280Z ##[command]"Get-BCArtifactUrl" 
2024-08-08T12:11:18.8601504Z Name                           Value                                                                                   
2024-08-08T12:11:18.8601728Z ----                           -----                                                                                   
2024-08-08T12:11:18.8601950Z select                         Weekly                                                                                  
2024-08-08T12:11:18.8602293Z type                           ***                                                                                 
2024-08-08T12:11:18.8602499Z country                        W1                                                                                      
2024-08-08T12:11:18.8602597Z 
2024-08-08T12:11:18.8602643Z 
2024-08-08T12:11:18.8602858Z  -select: Weekly -type: *** -country: W1
2024-08-08T12:11:58.2660479Z Downloading artifact /sandbox/24.4.22295.22576/w1
2024-08-08T12:11:58.3196226Z Downloading C:\Users\VssAdministrator\AppData\Local\Temp\c317bd5b-c378-47da-8b4b-24ecf209e13c.zip
2024-08-08T12:12:04.9767933Z Unpacking artifact to tmp folder using 7zip
2024-08-08T12:12:11.7795948Z Downloading platform artifact /sandbox/24.4.22295.22576/platform
2024-08-08T12:12:11.7826334Z Downloading C:\Users\VssAdministrator\AppData\Local\Temp\457e8833-318f-459e-8172-a4fd122dd38c.zip
2024-08-08T12:12:23.9793263Z Unpacking artifact to tmp folder using 7zip
2024-08-08T12:13:00.6166050Z Downloading Prerequisite Components
2024-08-08T12:13:00.6346809Z Downloading C:\bcartifacts.cache\sandbox\24.4.22295.22576\platform\Prerequisite Components\IIS URL Rewrite Module\rewrite_2.0_rtw_x64.msi
2024-08-08T12:13:00.7291060Z Downloading C:\bcartifacts.cache\sandbox\24.4.22295.22576\platform\Prerequisite Components\DotNetCore\DotNetCore.1.0.4_1.1.1-WindowsHosting.exe
2024-08-08T12:13:02.6214858Z *** Platform: C:\bcartifacts.cache\sandbox\24.4.22295.22576\platform
2024-08-08T12:13:02.6248817Z *** Localisation: C:\bcartifacts.cache\sandbox\24.4.22295.22576\w1
2024-08-08T12:13:02.7958633Z *** BC Management DLL: C:\bcartifacts.cache\sandbox\24.4.22295.22576\platform\ServiceTier\program files\Microsoft Dynamics NAV\240\Service\Management\Microsoft.Dynamics.Nav.Management.dll
2024-08-08T12:13:02.8090066Z *** Selected module: [C:\bcartifacts.cache\sandbox\24.4.22295.22576\platform\ServiceTier\program files\Microsoft Dynamics NAV\240\Service\Management\Microsoft.Dynamics.Nav.Management.dll]
2024-08-08T12:13:02.8101599Z *** Loading module: [C:\bcartifacts.cache\sandbox\24.4.22295.22576\platform\ServiceTier\program files\Microsoft Dynamics NAV\240\Service\Management\Microsoft.Dynamics.Nav.Management.dll]
2024-08-08T12:13:03.0555047Z *** BC Management DLL: C:\bcartifacts.cache\sandbox\24.4.22295.22576\platform\ServiceTier\program files\Microsoft Dynamics NAV\240\Service\Management\Microsoft.Dynamics.Nav.Management.dll
2024-08-08T12:13:03.0584297Z *** Selected module: [C:\bcartifacts.cache\sandbox\24.4.22295.22576\platform\ServiceTier\program files\Microsoft Dynamics NAV\240\Service\Management\Microsoft.Dynamics.Nav.Management.dll]
2024-08-08T12:13:03.0602306Z *** Loading module: [C:\bcartifacts.cache\sandbox\24.4.22295.22576\platform\ServiceTier\program files\Microsoft Dynamics NAV\240\Service\Management\Microsoft.Dynamics.Nav.Management.dll]
2024-08-08T12:13:03.0637062Z *** Load module [Microsoft.Dynamics.Nav.Apps.Management]
2024-08-08T12:13:03.2866401Z *** BC App Management DLL: not available
2024-08-08T12:13:03.2879682Z *** Batch Publish Extensions
2024-08-08T12:13:03.2890160Z   * Artifact Path  : D:\a\r1\a
2024-08-08T12:13:03.3071027Z *** Resolved Path  : D:\a\r1\a
2024-08-08T12:13:03.3080793Z *** Get Available Applications from [D:\a\r1\a].
2024-08-08T12:13:03.3176756Z *** Get [*.app] files recursive from [D:\a\r1\a]
2024-08-08T12:13:03.3198102Z *** Found [1] files.
2024-08-08T12:13:03.3208952Z *** Extract App detail from [D:\a\r1\a\_Noda.Automation.DynamicsExtensions\Noda business process automation\Noda - automation team_Noda.Automation.DynamicsExtensions_1.0.30719.0_onprem_24.4_dk.app]
2024-08-08T12:13:09.8379644Z 
2024-08-08T12:13:09.8380191Z 
2024-08-08T12:13:09.8380657Z ServerInstance                        : 
2024-08-08T12:13:09.8380917Z Brief                                 : 
2024-08-08T12:13:09.8381092Z Description                           : 
2024-08-08T12:13:09.8381339Z CompatibilityId                       : 0.0.0.0
2024-08-08T12:13:09.8381588Z PrivacyStatement                      : 
2024-08-08T12:13:09.8381831Z Eula                                  : 
2024-08-08T12:13:09.8389671Z Help                                  : 
2024-08-08T12:13:09.8390245Z Url                                   : 
2024-08-08T12:13:09.8390672Z Logo                                  : 
2024-08-08T12:13:09.8391014Z DeveloperTenantId                     : 
2024-08-08T12:13:09.8391419Z ScreenShots                           : {}
2024-08-08T12:13:09.8391877Z Dependencies                          : {Agreements, Nprogression, 1.0.0.0, False, Application, Microsoft, 24.0.0.0, 
2024-08-08T12:13:09.8392294Z                                         False}
2024-08-08T12:13:09.8392680Z Capabilities                          : {}
2024-08-08T12:13:09.8393098Z PackageId                             : c2d41ea8-4aee-4467-8313-86061377a79c
2024-08-08T12:13:09.8393531Z PublisherAzureActiveDirectoryTenantId : 
2024-08-08T12:13:09.8393911Z AppId                                 : 95ffab3e-3eb5-4b13-8907-f1dc89381d9d
2024-08-08T12:13:09.8394352Z Name                                  : Noda.Automation.DynamicsExtensions
2024-08-08T12:13:09.8394766Z Publisher                             : Noda - automation team
2024-08-08T12:13:09.8395439Z Version                               : 1.0.30719.0
2024-08-08T12:13:09.8395868Z ExtensionType                         : ModernDev
2024-08-08T12:13:09.8396264Z Scope                                 : Global
2024-08-08T12:13:09.8396597Z Tenant                                : 
2024-08-08T12:13:09.8396860Z 
2024-08-08T12:13:09.8397049Z 
2024-08-08T12:13:09.8397223Z 
2024-08-08T12:13:09.8397530Z 
2024-08-08T12:13:09.8543692Z *** Remove duplicate Apps
2024-08-08T12:13:09.8759773Z *** Installing Applications:
2024-08-08T12:13:09.8773371Z 
2024-08-08T12:13:09.8773748Z Publisher              Name                               Path                                                         
2024-08-08T12:13:09.8774493Z ---------              ----                               ----                                                         
2024-08-08T12:13:09.8774873Z Noda - automation team Noda.Automation.DynamicsExtensions D:\a\r1\a\_Noda.Automation.DynamicsExtensions\Noda busines...
2024-08-08T12:13:09.8775100Z 
2024-08-08T12:13:09.8775168Z 
2024-08-08T12:13:09.8775786Z 
2024-08-08T12:13:09.8815202Z *** Publish App [D:\a\r1\a\_Noda.Automation.DynamicsExtensions\Noda business process automation\Noda - automation team_Noda.Automation.DynamicsExtensions_1.0.30719.0_onprem_24.4_dk.app]
2024-08-08T12:13:09.9048880Z *** Upload extension [D:\a\r1\a\_Noda.Automation.DynamicsExtensions\Noda business process automation\Noda - automation team_Noda.Automation.DynamicsExtensions_1.0.30719.0_onprem_24.4_dk.app]
2024-08-08T12:13:09.9063628Z *** Get App Info
2024-08-08T12:13:10.8400727Z  * App.PackageID = c2d41ea8-4aee-4467-8313-86061377a79c
2024-08-08T12:13:10.8411639Z  * App.ID        = 95ffab3e-3eb5-4b13-8907-f1dc89381d9d
2024-08-08T12:13:10.8421706Z  * App.Name      = Noda.Automation.DynamicsExtensions
2024-08-08T12:13:10.8433574Z  * App.Publisher = Noda - automation team
2024-08-08T12:13:10.8444792Z  * App.Version   = 1.0.30719.0
2024-08-08T12:13:10.8540705Z *** Get Extension status
2024-08-08T12:13:10.8593271Z ##[command]Invoke-RestMethod -Method Get -Uri '***/v2.0/***/sandbox/api/microsoft/automation/v2.0/companies(5acb1e92-575c-ee11-8def-6045bd0fe76f)/extensions'
2024-08-08T12:13:11.8522101Z *** Replace PackageID in stream
2024-08-08T12:13:11.9467610Z *** Bytehack PackageID from [c2d41ea8-4aee-4467-8313-86061377a79c] to [9926ce42-e742-4d33-a2fb-f36ff1dcd516]
2024-08-08T12:13:12.0140469Z ##[command]Invoke-RestMethod -Method Patch -Uri '***/v2.0/***/sandbox/api/microsoft/automation/v2.0/companies(5acb1e92-575c-ee11-8def-6045bd0fe76f)/extensionUpload(0)/content'
2024-08-08T12:13:12.6273197Z ##[error]{
  "error": {
    "code": "BadRequest_NotFound",
    "message": "Bad Request - Error in query syntax."
  }
}
```

**Comment by waldo1001** [comment]
Can you try with apiversion v1.0?

**Comment by waldo1001** [comment]
Meanwhile, we found some changed documentation on the automation API's .. we'll try to fix the v2.

**Comment by waldo1001** [comment]
Fixed in v1.466

**Comment by VolkovYuVa** [comment]
@waldo1001, thank you, that solved this issue.
However, the publish still doesn't work due to another issue.
Now this issue appears:
```
2024-08-28T23:18:23.4984892Z *** New Cache Item
2024-08-28T23:18:23.5266705Z ##[command]"Get-BCArtifactUrl" 
2024-08-28T23:18:23.5267260Z Name                           Value                                                                                   
2024-08-28T23:18:23.5270032Z ----                           -----                                                                                   
2024-08-28T23:18:23.5270409Z select                         Weekly                                                                                  
2024-08-28T23:18:23.5270872Z type                           ***                                                                                 
2024-08-28T23:18:23.5271197Z country                        W1                                                                                      
2024-08-28T23:18:23.5271384Z 
2024-08-28T23:18:23.5271569Z 
2024-08-28T23:18:23.5271912Z  -select: Weekly -type: *** -country: W1
2024-08-28T23:18:50.6380485Z Downloading artifact /sandbox/24.4.22295.23242/w1
2024-08-28T23:18:50.6745988Z Downloading C:\Users\VssAdministrator\AppData\Local\Temp\b35bc8a7-e229-4831-8e51-166f7c4df41f.zip
2024-08-28T23:19:00.4745680Z Unpacking artifact to tmp folder using 7zip
2024-08-28T23:19:06.7454366Z Downloading platform artifact /sandbox/24.4.22295.23242/platform
2024-08-28T23:19:06.7480463Z Downloading C:\Users\VssAdministrator\AppData\Local\Temp\21e795ff-9610-47fb-9bf8-73efed021bc1.zip
2024-08-28T23:19:32.3771321Z Unpacking artifact to tmp folder using 7zip
2024-08-28T23:19:59.7251604Z Downloading Prerequisite Components
2024-08-28T23:19:59.7392161Z Downloading C:\bcartifacts.cache\sandbox\24.4.22295.23242\platform\Prerequisite Components\IIS URL Rewrite Module\rewrite_2.0_rtw_x64.msi
2024-08-28T23:19:59.8065861Z Downloading C:\bcartifacts.cache\sandbox\24.4.22295.23242\platform\Prerequisite Components\DotNetCore\DotNetCore.1.0.4_1.1.1-WindowsHosting.exe
2024-08-28T23:20:01.0500151Z *** Platform: C:\bcartifacts.cache\sandbox\24.4.22295.23242\platform
2024-08-28T23:20:01.0526117Z *** Localisation: C:\bcartifacts.cache\sandbox\24.4.22295.23242\w1
2024-08-28T23:20:01.1647615Z *** BC Management DLL: C:\bcartifacts.cache\sandbox\24.4.22295.23242\platform\ServiceTier\program files\Microsoft Dynamics NAV\240\Service\Management\Microsoft.Dynamics.Nav.Management.dll
2024-08-28T23:20:01.1755445Z *** Selected module: [C:\bcartifacts.cache\sandbox\24.4.22295.23242\platform\ServiceTier\program files\Microsoft Dynamics NAV\240\Service\Management\Microsoft.Dynamics.Nav.Management.dll]
2024-08-28T23:20:01.1763279Z *** Loading module: [C:\bcartifacts.cache\sandbox\24.4.22295.23242\platform\ServiceTier\program files\Microsoft Dynamics NAV\240\Service\Management\Microsoft.Dynamics.Nav.Management.dll]
2024-08-28T23:20:01.3941991Z *** BC Management DLL: C:\bcartifacts.cache\sandbox\24.4.22295.23242\platform\ServiceTier\program files\Microsoft Dynamics NAV\240\Service\Management\Microsoft.Dynamics.Nav.Management.dll
2024-08-28T23:20:01.3957821Z *** Selected module: [C:\bcartifacts.cache\sandbox\24.4.22295.23242\platform\ServiceTier\program files\Microsoft Dynamics NAV\240\Service\Management\Microsoft.Dynamics.Nav.Management.dll]
2024-08-28T23:20:01.3966635Z *** Loading module: [C:\bcartifacts.cache\sandbox\24.4.22295.23242\platform\ServiceTier\program files\Microsoft Dynamics NAV\240\Service\Management\Microsoft.Dynamics.Nav.Management.dll]
2024-08-28T23:20:01.4049568Z *** Load module [Microsoft.Dynamics.Nav.Apps.Management]
2024-08-28T23:20:01.5355933Z *** BC App Management DLL: not available
2024-08-28T23:20:01.5391928Z *** Publish Extension
2024-08-28T23:20:01.5399522Z   * Artifact Path  : D:\a\r1\a
2024-08-28T23:20:01.5406794Z   * Artifact Filter: *.app
2024-08-28T23:20:01.5480715Z *** Resolve App file in [D:\a\r1\a]
2024-08-28T23:20:01.5511034Z *** Resolved App File: [D:\a\r1\a\_Noda.Automation.DynamicsExtensions\Noda business process automation\Noda - automation team_Noda.Automation.DynamicsExtensions_1.0.32430.0_onprem_24.4_dk.app].
2024-08-28T23:20:01.5530766Z *** Resolved App Path: [D:\a\r1\a\_Noda.Automation.DynamicsExtensions\Noda business process automation\Noda - automation team_Noda.Automation.DynamicsExtensions_1.0.32430.0_onprem_24.4_dk.app]
2024-08-28T23:20:01.5673818Z *** Upload extension [D:\a\r1\a\_Noda.Automation.DynamicsExtensions\Noda business process automation\Noda - automation team_Noda.Automation.DynamicsExtensions_1.0.32430.0_onprem_24.4_dk.app]
2024-08-28T23:20:01.5684169Z *** Get App Info
2024-08-28T23:20:09.0760501Z  * App.PackageID = bbb6722f-7567-463f-bae0-f44ea4d2ce7c
2024-08-28T23:20:09.0781668Z  * App.ID        = 95ffab3e-3eb5-4b13-8907-f1dc89381d9d
2024-08-28T23:20:09.0808455Z  * App.Name      = Noda.Automation.DynamicsExtensions
2024-08-28T23:20:09.0830668Z  * App.Publisher = Noda - automation team
2024-08-28T23:20:09.0852648Z  * App.Version   = 1.0.32430.0
2024-08-28T23:20:09.0910245Z *** Get Extension status
2024-08-28T23:20:09.0952834Z ##[command]Invoke-RestMethod -Method Get -Uri '***/v2.0/***/sandbox/api/microsoft/automation/v2.0/companies(5acb1e92-575c-ee11-8def-6045bd0fe76f)/extensions'
2024-08-28T23:20:09.3704945Z *** Replace PackageID in stream
2024-08-28T23:20:09.5480073Z *** Bytehack PackageID from [bbb6722f-7567-463f-bae0-f44ea4d2ce7c] to [631502e9-39c1-4b30-a483-d1023f899f16]
2024-08-28T23:20:09.6120353Z *** Search existing Extension Upload
2024-08-28T23:20:09.6128079Z ##[command]Invoke-RestMethod -Method Get -Uri '***/v2.0/***/sandbox/api/microsoft/automation/v2.0/companies(5acb1e92-575c-ee11-8def-6045bd0fe76f)/extensionUpload'
2024-08-28T23:20:09.7065149Z 
2024-08-28T23:20:09.7065355Z 
2024-08-28T23:20:09.7068479Z @odata.etag                          : W/"JzIwOzExMjIzNDE1NTY4MDQ5NDgzNDI4MTswMDsn"
2024-08-28T23:20:09.7069002Z systemId                             : aca74900-e154-ef11-bfe2-000d3ad534d2
2024-08-28T23:20:09.7069355Z schedule                             : Current_x0020_version
2024-08-28T23:20:09.7069777Z schemaSyncMode                       : Add
2024-08-28T23:20:09.7070214Z extensionContent@odata.mediaEditLink : ***/v2.0/341adbdf-8878-4474-b82d-55481d6f3
2024-08-28T23:20:09.7070600Z                                        457/sandbox/api/microsoft/automation/v2.0/companies(5acb1e92-575c-ee11-8def-6045
2024-08-28T23:20:09.7071042Z                                        bd0fe76f)/extensionUpload(aca74900-e154-ef11-bfe2-000d3ad534d2)/extensionContent
2024-08-28T23:20:09.7071458Z extensionContent@odata.mediaReadLink : ***/v2.0/341adbdf-8878-4474-b82d-55481d6f3
2024-08-28T23:20:09.7071820Z                                        457/sandbox/api/microsoft/automation/v2.0/companies(5acb1e92-575c-ee11-8def-6045
2024-08-28T23:20:09.7072201Z                                        bd0fe76f)/extensionUpload(aca74900-e154-ef11-bfe2-000d3ad534d2)/extensionContent
2024-08-28T23:20:09.7072413Z 
2024-08-28T23:20:09.7072574Z 
2024-08-28T23:20:09.7072721Z 
2024-08-28T23:20:09.7072863Z 
2024-08-28T23:20:09.7125962Z *** Extension Upload (13308 bytes)
2024-08-28T23:20:09.7134993Z ##[command]Invoke-RestMethod -Method Patch -Uri '***/v2.0/***/sandbox/api/microsoft/automation/v2.0/companies(5acb1e92-575c-ee11-8def-6045bd0fe76f)/extensionUpload(aca74900-e154-ef11-bfe2-000d3ad534d2)/extensionContent'
2024-08-28T23:20:09.8138536Z *** Finish extension Upload
2024-08-28T23:20:09.8146612Z ##[command]Invoke-RestMethod -Method Post -Uri '***/v2.0/***/sandbox/api/microsoft/automation/v2.0/companies(5acb1e92-575c-ee11-8def-6045bd0fe76f)/extensionUpload(aca74900-e154-ef11-bfe2-000d3ad534d2)/Microsoft.NAV.upload'
2024-08-28T23:20:10.4731963Z ##[error]{"error":{"code":"Unknown","message":"The specified file is not an extension file. (Parameter 'stream')  CorrelationId:  2331d499-d430-49b6-b404-31079f7352b3."}}
2024-08-28T23:20:10.5363398Z ##[section]Finishing: ALOps Extension API
```

This artifact was previously built by a CI pipeline with the following yaml file:
```
# Starter pipeline
# Start with a minimal pipeline that you can customize to build and deploy your code.
# Add steps that build, run tests, deploy, and more:
# https://aka.ms/yaml

trigger: none

pool:
  vmImage: windows-latest

steps:
- task: ALOpsAppCompiler@2
  inputs:
    appversiontemplate: '1.0.*.0'
    appfilenametemplate: '%APP_PUBLISHER%_%APP_NAME%_%APP_VERSION%_%BC_TYPE%_%BC_VERSION%_%BC_COUNTRY%.app'
```
    
    I tried to upload exactly the same artifact manually and it worked.
    So then the artifact was definitely built correctly but for some reasons the publish function didn't work.
    
<img width="506" alt="image" src="https://github.com/user-attachments/assets/c4fc9785-08f8-4916-bd7c-30b45d39d8b2">

**Comment by VolkovYuVa** [comment]
Yes, it worked, thank you!
For some reasons it worked only after I merged the both pipelines.
